byron/chain/executable-spec/test/Cardano/AbstractSize/Properties.hs

```diff
@@ -0,0 +1,163 @@
+
+aTx :: Tx
+aTx = undefined
+
+aTxId :: TxId
+aTxId = TxId (hash aTx)
+
+aHeader :: BlockHeader
+aHeader = MkBlockHeader {
+          _bhPrevHash = (undefined :: Hash)
+        , _bhSlot = (undefined :: Slot)
+        , _bhIssuer = (undefined :: VKey)
+        , _bhSig = (undefined :: Sig Hash)
+        , _bhUtxoHash = (undefined :: Hash)
+        , _bhDlgHash = (undefined :: Hash)
+        , _bhUpdHash = (undefined :: Hash)
+     }
+
+aTxWits :: TxWits
+aTxWits =
+  let (in0,in1) = (TxIn aTxId 0, TxIn aTxId 1)
+      outs = []
+      wits = []
+   in TxWits (Tx [in0, in1] outs) wits
+
+aBody :: BlockBody
+aBody = BlockBody
+      { _bDCerts  = []
+      , _bUtxo  = [aTxWits, aTxWits]
+      , _bUpdProp = Nothing
+      , _bUpdVotes = []
+      , _bProtVer = ProtVer
+                      { _pvMaj = 0
+                      , _pvMin = 1
+                      , _pvAlt = 1
+                      }
+      }
+
+aBlock :: Block
+aBlock = Block { _bHeader = aHeader
+               , _bBody = aBody
+               }
+
+-- | A BlockHeader term has fixed typeReps
+exampleTypeRepsBlockHeader :: Assertion
+exampleTypeRepsBlockHeader =
+  (typeReps aHeader) @?= Seq.fromList [ typeOf (undefined::BlockHeader),
+        typeOf (undefined::Hash), typeOf (undefined::Hash),
+        typeOf (undefined::Hash), typeOf (undefined::Hash),
+        typeOf (undefined::Slot), typeOf (undefined::Word64),
+        typeOf (undefined::VKey), typeOf (undefined::Owner),
+        typeOf (undefined::Natural), typeOf (undefined::Sig Hash) ]
+
+-- | A BlockBody has variable typeReps, depending on the collections
+-- [DCert], [TxWits], [Vote] and [STag] (in UProp)
+--
+--   In this example, we can see the repetition of typeReps for 2 TxWits
+exampleTypeRepsBlockBody :: Assertion
+exampleTypeRepsBlockBody =
+     (typeReps aBody) @?=
+        Seq.fromList [
+            typeOf (undefined::BlockBody)
+          , typeOf (undefined::[DCert])
+          , typeOf (undefined::[TxWits]) ]
+        >< (typeReps aTxWits)
+        >< (typeReps aTxWits)
+        >< Seq.fromList [
+              typeOf (undefined::Maybe UProp), typeOf (undefined::[Vote])
+            , typeOf (undefined::ProtVer), typeOf (undefined::Natural)
+            , typeOf (undefined::Natural), typeOf (undefined::Natural) ]
+
+-- | The typeReps for a 'Block' is a combination of typeReps for
+-- the header and body in the block.
+exampleTypeRepsBlock :: Assertion
+exampleTypeRepsBlock =
+     (typeReps aBlock) @?= typeOf (undefined::Block)
+                           <| typeReps aHeader
+                           >< typeReps aBody
+
+--------------------------------------------------------------------------------
+-- Properties of abstractSize for Block/Header/Body
+--------------------------------------------------------------------------------
+
+-- | Make a singleton cost of "1" for the given term's type
+mkCost :: forall a. Typeable a => Map TypeRep Size
```

Comment:
If we would merge `Ledger.HasTypeReps.Properties` and `Cardano.AbstractSize.Properties` into a module we could remove this duplication.

Reply:
Forget what I said. I don't know whether it is sensible to add a module `Data.AbstractSize.Test` to the `small-steps` package with this function only. I guess for now we can leave this duplication. I'll leave this up to you :)